sci-biology/imagej/imagej-9999.ebuild

```diff
@@ -0,0 +1,131 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit git-r3 java-pkg-2 java-ant-2 desktop
+
+MY_PN="ij"
+IJ_PV="153" #plugins not currently available under 154
+
+DESCRIPTION="Image Processing and Analysis in Java"
+
+HOMEPAGE="
+	https://imagej.nih.gov/ij/
+	https://github.com/imagej
+"
+
+EGIT_REPO_URI="https://github.com/imagej/ImageJ"
+
+SRC_URI="
+	https://rsb.info.nih.gov/ij/images/ImageJ.png
+	plugins? ( https://wsr.imagej.net/distros/cross-platform/${MY_PN}${IJ_PV}.zip )"
+# plugins are under a different licenses and can be installed into user's $IJ_HOME/plugins
+#	plugins? ( http://rsb.info.nih.gov/ij/download/zips/${MY_PN}${IJ_PV}.zip )"
+
+KEYWORD="**"
+
+LICENSE="public-domain"
+
+SLOT="0"
+
+IUSE="doc plugins debug"
+
+RDEPEND="
+	>=virtual/jre-1.7:*
+	dev-java/java-config
+"
+DEPEND="${RDEPEND}
+	>=virtual/jdk-1.7:*
+"
+BDEPEND="
+	dev-java/ant-core
+	app-arch/unzip
+"
+
+IJ_S=${WORKDIR}/ImageJ
+
+src_prepare() {
+	cp "${DISTDIR}"/ImageJ.png "${WORKDIR}/${PN}.png" || die
+
+	if use plugins ; then
+		unzip "${DISTDIR}/${MY_PN}${IJ_PV}.zip" -d "${WORKDIR}"
```

Comment:
Instead of unzip (which is an external command and needs `|| die`. Use `unpack` which is the ebuild specific function for this.

Reply:
@AndrewAmmerlaan 

I can't seem to be able to get it to work with unpack.

Code in sci-biology/imagej/imagej-9999.ebuild

```
src_unpack(){
	if use plugins ; then
		unpack "${MY_PN}${IJ_PV}.zip"
	fi
}

src_prepare() {
	cp "${DISTDIR}"/ImageJ.png "${WORKDIR}/${PN}.png" || die

	if ! use debug ; then
		sed -i 's: debug="on">: debug="off">:' "${S}"/build.xml || die
	fi
	default
}
```

Fails as follow:

```
>>> Emerging (1 of 1) sci-biology/imagej-9999::localrepo
 * ImageJ.png BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                  [ ok ]
 * ij153.zip BLAKE2B SHA512 size ;-) ...                                                                                                                                                                                   [ ok ]
 * Using: openjdk-bin-17
>>> Unpacking source...
>>> Source unpacked in /var/tmp/portage/sci-biology/imagej-9999/work
 * ERROR: sci-biology/imagej-9999::localrepo failed (prepare phase):
 *   The source directory '/var/tmp/portage/sci-biology/imagej-9999/work/imagej-9999' doesn't exist
 * 
 * Call stack:
 *            ebuild.sh, line  780:  Called __ebuild_main 'prepare'
 *   phase-functions.sh, line 1072:  Called __dyn_prepare
 *   phase-functions.sh, line  400:  Called die
 * The specific snippet of code:
 *              die "The source directory '${S}' doesn't exist"
 * 
 * If you need support, post the output of `emerge --info '=sci-biology/imagej-9999::localrepo'`,
 * the complete build log and the output of `emerge -pqv '=sci-biology/imagej-9999::localrepo'`.
!!! When you file a bug report, please include the following information:
GENTOO_VM=openjdk-bin-17  CLASSPATH="" JAVA_HOME="/opt/openjdk-bin-17.0.6_p10"
JAVACFLAGS="-source 1.7 -target 1.7" COMPILER=""
and of course, the output of emerge --info =imagej-9999
 * The complete build log is located at '/var/tmp/portage/sci-biology/imagej-9999/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/sci-biology/imagej-9999/temp/environment'.
 * Working directory: '/var/tmp/portage/sci-biology/imagej-9999/empty'
 * S: '/var/tmp/portage/sci-biology/imagej-9999/work/imagej-9999'

>>> Failed to emerge sci-biology/imagej-9999, Log file:

```

To be noted, the src_unpack is causing the git url NOT to be fetched at all, and therefore the " The source directory '/var/tmp/portage/sci-biology/imagej-9999/work/imagej-9999' doesn't exist". (True, I verified and it didnt')

The unpacking worked ok.

Not sure if I am missing something. 
Or if you want me to push the changes as is, not working for you to look at it,
Or if in this case the unzip use is acceptable and we add || die

Regards